bundle/jsky.app.ot/src/main/scala/edu/gemini/catalog/ui/QueryResultsFrame.scala

```diff
@@ -335,8 +397,10 @@ object QueryResultsFrame extends Frame with PreferredSizeFrame {
         }
     }
 
-    lazy val objectName = new TextField("") with SelectOnFocus {
-      val foregroundColor = UIManager.getColor("TextField.foreground")
+    lazy val objectName: TextField with SelectOnFocus {
+      def resetName(n: String): Unit
+    } = new TextField("") with SelectOnFocus {
```

Comment:
All these refinements are slightly sketchy because it's so easy to crush them down to non-refined types, but I guess if it works here it's fine. The alternative would be to declare a bunch of single-use traits I guess.